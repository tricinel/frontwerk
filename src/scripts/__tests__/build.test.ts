/* eslint-disable global-require */
import cases from 'jest-in-case';
import jestSerializerPath from 'jest-serializer-path';

import {
  unquoteSerializer,
  winPathSerializer
} from '../../helpers/serializers';

expect.addSnapshotSerializer(unquoteSerializer);
expect.addSnapshotSerializer(winPathSerializer);
expect.addSnapshotSerializer(jestSerializerPath);

jest.mock('consola');

const testCases = [
  {
    name: 'calls node with default args'
  },
  {
    name: 'calls node with --bundle',
    args: ['--bundle']
  },
  {
    name: 'calls node with --pack',
    args: ['--pack']
  }
];

const testFn = ({ args = [] }) => {
  const crossSpawn = require('cross-spawn');
  const syncSpy = jest
    .spyOn(crossSpawn, 'sync')
    .mockImplementation(() => ({ status: 0 }));
  const exitSpy = jest.spyOn(process, 'exit');
  const { argv: originalArgv } = process;

  try {
    // Tests
    process.argv = ['node', '../build', ...args];
    require('../build');

    expect(syncSpy).toHaveBeenCalledTimes(1);
    const [firstCall] = syncSpy.mock.calls;
    const [script, calledArgs] = firstCall as [string, string[]];
    expect([script, ...calledArgs].join(' ')).toMatchSnapshot();
  } finally {
    // We reset everything afterEach
    exitSpy.mockRestore();
    syncSpy.mockRestore();
    process.argv = originalArgv;
    jest.resetModules();
  }
};

cases('build', testFn, testCases);
