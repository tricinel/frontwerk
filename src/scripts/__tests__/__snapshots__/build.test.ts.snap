// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build calls node with --bundle 1`] = `concurrently --names esm,cjs,umd,umd.min "cross-env BUILD_ROLLUP=true BUILD_FORMAT=esm BUILD_MINIFY=false NODE_ENV=development  rollup --config" "cross-env BUILD_ROLLUP=true BUILD_FORMAT=cjs BUILD_MINIFY=false NODE_ENV=development  rollup --config" "cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=false NODE_ENV=development  rollup --config --sourcemap" "cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=true NODE_ENV=production  rollup --config --sourcemap"`;

exports[`build calls node with --pack 1`] = `babel --out-dir dist --copy-files --ignore __tests__,__mocks__ --presets <PROJECT_ROOT>/src/config/babelrc.js src --pack`;

exports[`build calls node with default args 1`] = `babel --out-dir dist --copy-files --ignore __tests__,__mocks__ --presets <PROJECT_ROOT>/src/config/babelrc.js src`;
